app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -61,6 +61,30 @@ def update_os_from_facts
         scoped_search :relation => :host_traces, :on => :application, :complete_value => true, :rename => :trace_app, :only_explicit => true
         scoped_search :relation => :host_traces, :on => :app_type, :complete_value => true, :rename => :trace_app_type, :only_explicit => true
         scoped_search :relation => :host_traces, :on => :helper, :complete_value => true, :rename => :trace_helper, :only_explicit => true
+
+        scoped_search relation: :pools, on: :pools_expiring_in_days, ext_method: :find_with_expiring_pools, only_explicit: true
+
+        def self.find_with_expiring_pools(_key, _operator, days_from_now)
+          minimum_pool_date = (Date.today + days_from_now.to_i.days).strftime('%Y-%m-%d')
+          host_ids = self.joins(:pools).where(["katello_pools.end_date < ?", minimum_pool_date]).distinct.ids
```

Comment:
Based on my recommendation of using a scope I think you could call that here ie `pools.expiring_in_days.distinct.ids` Sorry if this feedback is all irrelevant because of scoped search, but I *think* it should be OK